src/model/mod.rs

```diff
@@ -183,6 +186,29 @@ impl Serialize for CapabilityConfig {
     }
 }
 
+impl Hash for CapabilityConfig {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        match self {
+            CapabilityConfig::Json(v) => {
+                //NOTE(@ahmedtadde): By default, Serde uses BtreeMap for JSON map objects which is good for determinism.
+                //However, for arrays, ordering is not guaranteed.
+                //In general, unless the json values are strictly equal (ordering, casing, white spaces, etc...), the hash will be different. This is something to be mindful of when using this hash function.
+                let json = match serde_json::to_string(v) {
```

Comment:
If we do need to keep this (see other comment), I worry about having to serialize every time we want to hash. I know we only do it once, but it feels like with that overhead it doesn't belong in a `Hash` implementation so much as written into something the a base64 encoder and hashing that way

Reply:
[is this change inline with what you have in mind](https://github.com/wasmCloud/wadm/pull/185/commits/cab231f5402b1b9c95e16e8925b7f3ac0127ea64)